tests/backup/backup_helper.go

```diff
@@ -286,6 +288,29 @@ func CreateBackup(backupName string, clusterName string, bLocation string, bLoca
 	if err != nil {
 		return err
 	}
+
+	err = backupSuccessCheck(backupName, orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
+	if err != nil {
+		return err
+	}
+	log.Infof("Backup [%s] created successfully", backupName)
+	return nil
+}
+
+// CreateBackupWithCRValidation creates backup and checks for success along with Backup CR Validation
+func CreateBackupWithCRValidation(backupName string, clusterName string, bLocation string, bLocationUID string,
+	namespaces []string, labelSelectors map[string]string, orgID string, uid string, preRuleName string,
+	preRuleUid string, postRuleName string, postRuleUid string, ctx context.Context) error {
+	_, err := CreateBackupByNamespacesWithoutCheck(backupName, clusterName, bLocation, bLocationUID, namespaces, labelSelectors, orgID, uid, preRuleName, preRuleUid, postRuleName, postRuleUid, ctx)
+	if err != nil {
+		return err
+	}
+
+	err = ValidateBackupCRs(backupName, clusterName, orgID, uid, ctx, namespaces)
```

Comment:
If we can just pass `BackupObject` and `ctx` we should be good here. We should be able to get all the other details from `CreateBackup` itself.

Reply:
Refactored as discussed